swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/Subscription.java

```diff
@@ -43,21 +45,69 @@ public class Subscription {
   @NotNull @NotEmpty private String id; // required
 
   /**
-   * Enables capability to inherit billing model information from their parent subscription Unused
-   * prior to https://issues.redhat.com/browse/BIZ-629
+   * Enables capability to inherit billing model information from their parent subscription. Unused
+   * prior to <a href="https://issues.redhat.com/browse/BIZ-629">BIZ-629</a>
    */
   private String parentSubscription;
 
   /**
    * defines an "in-the-box" subscription. Considered included from both usage and capacity
    * perspectives.
    */
-  private List<String> includedSubscriptions;
+  private List<String> includedSubscriptions = new ArrayList<>();
 
   private Fingerprint fingerprint;
-  private List<Variant> variants;
+  private List<Variant> variants = new ArrayList<>();
   private BillingWindow billingWindow;
   private String serviceType;
-  private List<Metric> metrics;
+  private List<Metric> metrics = new ArrayList<>();
   private Defaults defaults;
+
+  public static Optional<Subscription> findByServiceType(String serviceType) {
+    return SubscriptionConfigRegistry.getInstance().lookupByServiceType(serviceType);
+  }
+
+  public static Optional<Subscription> findByArch(String arch) {
+    return SubscriptionConfigRegistry.getInstance().lookupByFingerprintArch(arch);
+  }
+
+  public List<String> getMetricIds() {
+    return this.getMetrics().stream()
+        .map(Metric::getId)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  public Optional<Metric> getMetric(String metricId) {
+    return this.getMetrics().stream()
+        .filter(Objects::nonNull)
+        .filter(x -> Objects.equals(x.getId(), metricId))
+        .findFirst();
+  }
+
+  public static Optional<Subscription> findById(String id) {
+    return SubscriptionConfigRegistry.getInstance().getSubscriptions().stream()
+        .filter(subscription -> Objects.equals(subscription.getId(), id))
+        .findFirst();
+  }
+
+  public static List<String> getAllServiceTypes() {
+    return SubscriptionConfigRegistry.getInstance().getAllServiceTypes();
+  }
+
+  public boolean isPrometheusEnabled() {
+
+    return this.getMetrics().stream().anyMatch(metric -> Objects.nonNull(metric.getPrometheus()));
+  }
+
+  public String getFinestGranularity() {
+
+    return this.isPrometheusEnabled() ? "HOURLY" : "DAILY";
+  }
+
+  public List<String> getSupportedGranularity() {
```

Comment:
Could we also change in the Jira card the method name `getSupportedGranularity` instead of `supportsGranularity`? I don't mind if we want to keep it the way it is currently in Jira.

Reply:
Done - renamed in the JIRA card description